docs/source/workflow/generate-types.mdx

```diff
@@ -0,0 +1,219 @@
+}
+
+type AddBookMutationResponse {
+  code: String!
+  success: Boolean!
+  message: String!
+  book: Book
+}
+
+type Mutation {
+  addBook(title: String, author: String): AddBookMutationResponse
+}
+```
+
+If you moved your schema into a `.graphql` file, update your imports to ensure you're still properly passing your schema to your server. In the file where you create your server, you can read in your schema using `readFileSync` from the `fs` package:
+
+```ts title="src/index.ts"
+// ...other imports
+import { readFileSync } from 'fs';
+import path, { dirname } from 'path';
+import { fileURLToPath } from 'url';
+
+// If you are using ESM module syntax you can redefine `__dirname` as 
+// shown below. If you are using CJS syntax you can use `__dirname` directly.
+const __dirname = dirname(fileURLToPath(import.meta.url));
+const schemaFile = path.join(__dirname, '../src/schema.graphql');
+// Once we have our schema's file path, read in our schema. 
+const typeDefs = readFileSync(schemaFile, { encoding: 'utf-8' });
+
+const server = new ApolloServer<MyContext>({
+  typeDefs,
+  resolvers,
+});
+
+// ... start our server
+```
+
+Restart your server to ensure it can find and use your schema and that everything works as expected. Next, we'll install the packages we need to generate types automatically based on our schema.
+
+### Installing and configuring dependencies
+
+Run the following command to install the `@graphql-codegen/cli`, `@graphql-codegen/typescript`, and `@graphql-codegen/typescript-resolvers` packages into your project's dev dependencies:
+
+```bash
+npm install -D @graphql-codegen/cli @graphql-codegen/typescript @graphql-codegen/typescript-resolvers
+```
+
+> For more information on each package above, check out the [GraphQL Code Generator](https://www.the-guild.dev/graphql/codegen/plugins) docs.
+
+Next, we'll set up a configuration file to tell GraphQL Code Generator where and how to generate types. You can do this by manually creating a `codegen.yml` file or by using the following command, which walks you through the process:
+
+```bash
+npx graphql-code-generator init
+```
+
+Below is an example of a `codegen.yml` file:
+
+```yaml
+# This configuration file tells GraphQL Code Generator how
+# to generate types based on our schema.
+schema: "./schema.graphql"
+generates:
+  # Specify where our generated types should live.
+  ./src/__generated__/resolvers-types.ts:
+    plugins:
+      - "typescript"
+      - "typescript-resolvers"
+    config:
+      useIndexSignature: true
+      # More on this below!
+      contextType: ../index#MyContext
+```
+
+> [See the docs](https://www.the-guild.dev/graphql/codegen/plugins/typescript-resolvers#usage-example) for more information on the above configuration options.
+
+Finally, we recommend adding helpful scripts to your `package.json` file to ensure your TS types are regularly generated:
+
+```json
+{
+// ...
+  "scripts": {
+    "compile": "graphql-codegen --config codegen.yml && tsc",
```

Comment:
Having a separate "generate-types" script might be nice here (which the "compile" script uses).

Reply:
Good call! Replaced with:

```json
  "scripts": {
    "generate": "graphql-codegen --config codegen.yml",
    "compile": "npm run generate && tsc",
    "postinstall": "npm run compile",
    "start": "npm run compile && node ./dist/index.js",
  },
```